test/unit/range/view/view_concept_check.hpp

```diff
@@ -0,0 +1,295 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Tobias Loka <LokaT AT rki.de>
+ * \brief Provides functions to test the fulfillment of range concepts.
+ */
+
+#include <gtest/gtest.h>
+
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/std/ranges>
+
+using namespace seqan3;
+
+namespace seqan3::test 
+{
+
+//!\brief Enum for the different types of range concepts.
+enum ConceptType : uint8_t
+{
+    Input,
+    Forward,
+    Bidirectional,
+    RandomAccess,
+    Contiguous,
+    Viewable,
+    View,
+    Sized,
+    Common,
+    Output,
+    ConstIterable
+};
+
+/*!\brief Convert a ConceptType object to std::string for output.
+ * \tparam conc The range concept to convert to string as seqan3::test::ConceptType object.
+ * \return The concept type as std::string.
+ *
+ *\details
+ * 
+ * The return string is not a 1:1 conversion of the enum type name but rather the full name
+ * of the range concept it stands for.
+ */
+std::string to_string(ConceptType const conc)
+{
+    switch(conc)
+    {
+        case Input:
+            return "InputRange";
+        case Forward:
+            return "ForwardRange";
+        case Bidirectional:
+            return "BidirectionalRange";
+        case RandomAccess:
+            return "RandomAccessRange";
+        case Contiguous:
+            return "ContiguousRange";
+        case Viewable:
+            return "ViewableRange";
+        case View:
+            return "View";
+        case Sized:
+            return "SizedRange";
+        case Common:
+            return "CommonRange";
+        case Output:
+            return "OutputRange";
+        case ConstIterable:
+            return "const_iterable_concept";
+        default:
+            return "Undefined";
+    }
+}
+
+/*!\brief Expression to check the fulfillment of a range concept.
+ * \tparam conc    The range concept to check as seqan3::test::ConceptType object.
+ * \return True if TYPE fulfills conc. False if not.
+ * \throws std::logic_error    Throws logic error if the concept type given (concept) is unknown.
+ *
+ * ### Example
+ * using namespace seqan3::test;
```

Comment:
I'm not sure if we need snippets for this, but you could wrap it in a 
```console
```cpp
source code
\```
``` 
block (without the \)